addons/account/models/account_bank_statement.py

```diff
@@ -891,6 +891,10 @@ def process_reconciliation(self, counterpart_aml_dicts=None, payment_aml_rec=Non
             # Create the move
             self.sequence = self.statement_id.line_ids.ids.index(self.id) + 1
             move_name = (self.statement_id.name or self.name) + "/" + str(self.sequence)
+            # Ensure no duplicate is created
+            if self.env['account.move'].search([('name', '=', move_name)]):
+                count = self.env['account.move'].search_count([('name', '=like', move_name + '/%')])
+                move_name = move_name + '/' + str(count + 1)
```

Comment:
What would be the advantage to do: `move_name = u'{}{}{:d}'.format(move_name, u'/', count + 1)` ?